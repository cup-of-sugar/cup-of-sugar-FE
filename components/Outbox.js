import React from "react";
import {
  ScrollView,
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  Image
} from "react-native";
import { SearchResult } from "./SearchResult";
import Colors from "../constants/Colors";
import { gql } from "apollo-boost";
import { useQuery } from "@apollo/react-hooks";
import borrower from "../assets/images/borrower.png";
import lender from "../assets/images/lender.png";

export const SENT_MESSAGES = gql`
  query UserOutbox($userId: ID!) {
    userOutbox(userId: $userId) {
      title
      body
      recipient {
        firstName
        email
      }
    }
  }
`;

export default function Outbox(props) {
  const navigation = props.navigation;
  const userId = props.userId;

  let { loading, error, data } = useQuery(SENT_MESSAGES, {
    variables: {
      userId
    },
    fetchPolicy: "network-only"
  });

  if (loading) return <Text style={styles.loadingText}>Loading...</Text>;

  if (error) return <Text style={styles.errorText}>No messages found!</Text>;

  if (data) {
    return data.userOutbox.length ? (
      data.userOutbox.map(mail => {
        return (
          <TouchableOpacity
            key={mail.title}
            style={styles.message}
            onPress={() =>
              navigation.navigate("Message", {
                message: mail,
                userId: props.userId
              })
            }
          >
            <Text style={styles.messageText}>
              To: {mail.recipient.firstName}
            </Text>
            <Image source={borrower} style={styles.icon} />
          </TouchableOpacity>
        );
      })
    ) : (
      <Text style={styles.errorText}>No messages found!</Text>
    );
  }
}

const styles = StyleSheet.create({
  message: {
    alignSelf: "center",
    backgroundColor: "#fff",
    borderRadius: 10,
    borderWidth: 1,
    borderColor: "#fff",
    flex: 0,
    flexDirection: "row",
    height: 70,
    marginHorizontal: 40,
    marginVertical: 10,
    paddingVertical: 20,
    paddingHorizontal: 15,
    width: 350
  },
  messageText: {
    fontSize: 20,
    fontWeight: "bold",
    color: Colors.darkBlue,
    paddingRight: 10
  },
  icon: {
    height: 25,
    width: 115
  },
  loadingText: {
    alignSelf: "center",
    fontSize: 20,
    fontWeight: "bold",
    margin: 10
  },
  errorText: {
    alignSelf: "center",
    color: "red",
    fontSize: 20,
    fontWeight: "bold",
    margin: 10,
    marginTop: 50
  }
});
