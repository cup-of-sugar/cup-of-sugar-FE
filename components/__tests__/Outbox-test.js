import React from "react";
import Outbox, { SENT_MESSAGES } from "../Outbox";

import { MockedProvider } from "@apollo/react-testing";
import ReactTestRenderer from "react-test-renderer";

it("should render component", async () => {
  const mocks = [
    {
      request: {
        query: SENT_MESSAGES,
      },
      result: {
        data: {
          userOutbox: [
            {
              title: "Tigers",
              body: "Still waiting on tigers",
              userId: "1",
              recipient: {
                id: "2",
                email: "carole@tigers.com",
                firstName: "Carole",
                lastName: "Baskin",
                __typename: "UserType",
              },
              __typename: "MessageType",
            },
          ],
        },
      },
    },
  ];

  const props = {
    navigation: { navigate: jest.fn() },
    userId: "2",
  };
  let tree = await ReactTestRenderer.create(
    <MockedProvider mocks={mocks}>
      <Outbox {...props} />
    </MockedProvider>
  );
  expect(tree).toMatchSnapshot();
});
