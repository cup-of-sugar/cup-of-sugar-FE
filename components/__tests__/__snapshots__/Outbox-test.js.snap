// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<Text
  style={
    Object {
      "alignSelf": "center",
      "fontSize": 20,
      "fontWeight": "bold",
      "margin": 10,
    }
  }
>
  Loading...
</Text>
`;
